### PR TITLE
✨ Source generate the ApplicationDbContext implementations

### PR DESCRIPTION
**Pull request by jscarle**
When dealing with Entity Framework in the context of a Clean Architecture application, over time, as the number of DbSets increase, the constant back and forth between the interface and the implementation can become rather tedious and is subject to implementations being left in the DbContext which are no longer used by the interface.

I wrote a source generator that will generate the implementations as a partial of the DbContext based on the properties of the interface: https://github.com/jscarle/GeneratedEntityFramework

This pull request removes the `TodoItems` DbSet implementation from the `ApplicationDbContext` and replaces it with the source generated implementation.

### CODE REVIEW

**Comment by Hona** [comment]
cc: @SSWConsulting/ssw-cleanarchitecture 
(Checked by @matt-goldman @christoment @william-liebenberg via Teams Chat)
**Hi @jscarle,**
 
Thank you for your effort in preparing this pull request.
 
Adding new dependencies is a decision that needs to be carefully measured and needs to solve an identified problem.
 
After some discussion with the team, we've decided not to merge this for a couple reasons:
 
- Added 'magic': developers are familiar with DbContext & having DbSet<> in the class. It is another thing to learn as a difference from the typical pattern.
- Adding DbSets is not really an issue for us. Adding a new table is only a few lines of boilerplate code. While it might be verbose, it is not complex.
 
In summary, while this is a neat idea, it doesn't align with this template.
 
We appreciate the attempt at improving DX, and will be delighted to review any future contributions.

Regards,
SSW SAW Team
